File: src/Veilora.Api/ClientApp/src/types/relationship.ts
export type RelationshipType = 'Spouse' | 'Partner' | 'Godparent' | 'Guardian' | 'CloseFriend';

export interface RelationshipDto {
  id: string;
  character1Id: string;
  character2Id: string;
  relationshipType: RelationshipType;
  startDate?: string;
  endDate?: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRelationshipDto {
  character1Id: string;
  character2Id: string;
  relationshipType: RelationshipType;
  startDate?: string;
  endDate?: string;
  notes?: string;
}

File: src/Veilora.Domain/Entities/Relationship.cs
using Veilora.Domain.Common;
using Veilora.Domain.Enums;

namespace Veilora.Domain.Entities;

public class Relationship : BaseEntity
{
    public Guid Character1Id { get; set; }
    public Guid Character2Id { get; set; }
    public RelationshipType RelationshipType { get; set; }
    public DateOnly? StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public string? Notes { get; set; }

    public Character Character1 { get; set; } = null!;
    public Character Character2 { get; set; } = null!;
}
